### PR TITLE
Light refactor

### PR DESCRIPTION
**Pull request by QnnOkabayashi**
This PR brings some small QOL improvements.
* Changed the `INFO` level icon from `＿` to `ｉ`. I thought the underscore looked out of place, and the `ｉ` is still unintrusive. It's still a full width character, so everything is still nicely aligned for pretty printing.
* Cleaned up the serialization of trees by inlining `Shared`'s fields and moving the `Span`'s children field to the end, so the inner spans are the last part of a span to print. Previously, the inner and total duration's would print after the children, which could be hundreds of lines down if the tree was deep.
* Added a `ForestLayer::sink` constructor for a layer that doesn't process trace data once a span closes. This is useful for testing setting and retrieving `Uuid`'s, but without wanting to do extra work.
* Added a dev-dep to fix a doc test that depended on `tracing_subscriber::EnvFilter`, which is behind a feature gate.
* Organized the tests so they're no longer in one giant file.

### CODE REVIEW

File: tracing-forest/src/layer/mod.rs [hunk 1]
```diff
@@ -356,9 +360,9 @@ where
 /// ```
 /// Produces the the output:
 /// ```log
-/// INFO     ＿ [info]: Hello, world!
+/// INFO     ｉ [info]: Hello, world!
 /// INFO     my_span [ 26.0µs | 100.000% ]
-/// INFO     ┕━ ＿ [info]: Relevant information
+/// INFO     ┕━ ｉ [info]: Relevant information
```
Comment (Firstyear) [inline tracing-forest/src/layer/mod.rs:365]: I still say this should be just two spaces.

File: tracing-forest/src/printer/pretty.rs [hunk 2]
```diff
@@ -28,22 +28,22 @@ use std::fmt::{self, Write};
...
-/// INFO     ┝━ server::internal_search [ 6.98ms | 31.88% / 93.47% ]
-/// INFO     │  ┝━ __ [filter.info]: Some filter info...
-/// INFO     │  ┝━ server::search [ 4.59ms | 0.81% / 61.41% ]
-/// INFO     │  │  ┝━ be::search [ 4.51ms | 0.40% / 60.31% ]
-/// INFO     │  │  │  ┕━ be::search -> filter2idl [ 4.48ms | 22.40% / 59.91% ]
-/// INFO     │  │  │     ┝━ be::idl_arc_sqlite::get_idl [ 571µs | 7.64% ]
-/// INFO     │  │  │     │  ┕━ ＿ [filter.info]: Some filter info...
-/// INFO     │  │  │     ┕━ be::idl_arc_sqlite::get_idl [ 2.23ms | 29.85% ]
-/// ERROR    │  │  │        ┝━ 🚨 [admin.error]: Oh no, an admin error occurred :(
+/// INFO     try_from_entry_ro [ 324µs | 8.47% / 100.00% ]
```
Comment (Firstyear) [inline tracing-forest/src/printer/pretty.rs:31]: Lol, I love the kanidm example

**Review by Firstyear — APPROVED** [review]
One minor comment, but as always looks good!

File: tracing-forest/src/layer/mod.rs (hunk 1 above)
Comment (QnnOkabayashi) [reply tracing-forest/src/layer/mod.rs:365]: I guess it doesn't really matter that much, but I'm curious why `ｉ` isn't as good at two spaces? I think it aligns with info better and is equally unobtrusive.

**Pull request merged**